exonum-java-binding/common/src/main/java/com/exonum/binding/common/proofs/map/UncheckedFlatMapProof.java

```diff
@@ -249,35 +257,53 @@ private CheckedMapProof checkProof() {
     return commonPrefix;
   }
 
-  private static HashCode getEmptyProofListHash() {
-    return HashCode.fromBytes(new byte[Hashing.DEFAULT_HASH_SIZE_BYTES]);
+  private static HashCode getEmptyProofMapHash() {
+    return HASH_FUNCTION.newHasher()
+        .putByte(MAP_NODE_PREFIX)
+        .putBytes(new byte[Hashing.DEFAULT_HASH_SIZE_BYTES])
+        .hash();
   }
 
-  private static HashCode getSingleEntryRootHash(MapProofEntry entry) {
-    return getSingleEntryRootHash(entry.getDbKey(), entry.getHash());
+  private static HashCode getSingleEntryProofMapHash(MapProofEntry proofEntry) {
+    HashCode entryRootHash = getSingleEntryRootProofMapHash(proofEntry.getDbKey(),
+        proofEntry.getHash());
+    return getProofMapHash(entryRootHash);
   }
 
-  private static HashCode getSingleEntryRootHash(MapEntry<ByteString, ByteString> entry) {
-    DbKey dbKey = DbKey.newLeafKey(entry.getKey());
-    HashCode valueHash = HASH_FUNCTION.hashByteString(entry.getValue());
-    return getSingleEntryRootHash(dbKey, valueHash);
+  private static HashCode getSingleEntryProofMapHash(MapEntry<ByteString, ByteString> mapEntry) {
+    DbKey dbKey = DbKey.newLeafKey(mapEntry.getKey());
+    HashCode valueHash = getLeafEntryHash(mapEntry.getValue());
+    HashCode entryRootHash = getSingleEntryRootProofMapHash(dbKey, valueHash);
+    return getProofMapHash(entryRootHash);
   }
 
-  private static HashCode getSingleEntryRootHash(DbKey key, HashCode valueHash) {
+  private static HashCode getProofMapHash(HashCode rootHash) {
+    return HASH_FUNCTION.newHasher()
+        .putByte(MAP_NODE_PREFIX)
+        .putObject(rootHash, hashCodeFunnel())
+        .hash();
+  }
+
+  private static HashCode getSingleEntryRootProofMapHash(DbKey key, HashCode valueHash) {
```

Comment:
Too long name. `getSingleEntryRootHash/MerkleRoot` (since we are inside MapProof class)?

Reply:
Also, why doesn't the code use the newer 'merkleRoot' term?